frontend/src/__tests__/api/authors.test.js

```diff
@@ -0,0 +1,124 @@
+  addAuthorMutation
+} from '../../../api/authors';
+import {
+  aBookQuery,
+  addBookMutation,
+  allBooksQuery,
+  removeBookMutation,
+  useAddBook,
+  useGetBook,
+  useGetBooks,
+  useRemoveBook
+} from '../../../api/books';
+import { MockedProvider } from '@apollo/client/testing';
+
+describe('api Authors test', () => {
+  it('should get all authors', async () => {
+    const getAuthorsMock = {
+      request: {
+        query: allAuthorsQuery,
+        variables: {}
+      },
+      result: {
+        data: {
+          getAuthors: [
+            {
+              id: '1',
+              firstName: 'J.K.',
+              lastName: 'Rowling'
+            },
+            {
+              id: '2',
+              firstName: 'Dan',
+              lastName: 'Gookin'
+            }
+          ]
+        }
+      }
+    };
+    const wrapper = ({ children }) => (
+      <MockedProvider mocks={[getAuthorsMock]} addTypename={false}>
+        {children}
+      </MockedProvider>
+    );
+
+    const { result } = renderHook(() => useGetAuthors(), { wrapper });
+    await waitFor(() => expect(result.current.authors.length).not.toBe(0));
+    expect(result.current.authorsError).toBeFalsy();
+    expect(result.current.authorsLoading).toBeFalsy();
+  });
+  it('should get an author', async () => {
+    const getAuthorMOCK = {
+      request: {
+        query: anAuthorQuery,
+        variables: { id: '1' }
+      },
+      result: {
+        data: {
+          getAuthor: {
+            id: '1',
+            firstName: 'J.K.',
+            lastName: 'Rowling',
+            books: [
+              {
+                id: '1',
+                title: 'Harry Potter'
+              }
+            ]
+          }
+        }
+      }
+    };
+    const wrapper = ({ children }) => (
+      <MockedProvider mocks={[getAuthorMOCK]} addTypename={false}>
+        {children}
+      </MockedProvider>
+    );
+
+    const { result } = renderHook(() => useGetAuthor('1'), { wrapper });
+    await waitFor(() => expect(result.current.authorLoading).toBeFalsy());
+    expect(JSON.stringify(result.current.author)).not.toBe('{}');
+  });
+  it('should add an author', async () => {
+    const addAuthorMock = {
+      request: {
+        query: addAuthorMutation,
+        variables: {
+          firstName: 'Jim',
+          lastName: 'Bean'
+        }
+      },
+      result: {
+        data: {
+          addAuthor: {
+            id: '1',
+            firstName: 'Jim',
+            lastName: 'Bean'
+          }
+        }
+      }
+    };
+    const wrapper = ({ children }) => (
+      <MockedProvider mocks={[addAuthorMock]} addTypename={false}>
+        {children}
+      </MockedProvider>
+    );
+
+    const { result } = renderHook(() => useAddAuthor(), { wrapper });
+
+    await waitFor(() => expect(result.current.addAuthorLoading).toBe(false));
+    let authorInfo;
+    await act(async () => {
+      authorInfo = await result.current.addAuthor('Jim', 'Bean');
+    });
```

Comment:
I think there might be some other way we can do this.

Reply:
Unsure what other way would be. I can reduce down to:
    await act(async () => {
      const authorInfo = await result.current.addAuthor('Jim', 'Bean');
      expect(authorInfo.data.addAuthor.id).toBe('1');
    });
    
    but I'm not sure that this is what you are driving at?

Reply:
You are incorrectly using act/await and async here.